lib/ronin/recon/builtin/api/zoom_eye.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+#
+# ronin-recon - A micro-framework and tool for performing reconnaissance.
+#
+# Copyright (c) 2023-2024 Hal Brodigan (postmodern.mod3@gmail.com)
+#
+# ronin-recon is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ronin-recon is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with ronin-recon.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+require 'ronin/recon/worker'
+
+require 'async/http/internet'
+
+module Ronin
+  module Recon
+    module API
+      #
+      # A recon worker that queries https://api.zoomeye.hk/domain/search
+      # and returns subdomain and ip addresses for a given domain
+      #
+      class ZoomEye < Worker
+
+        register 'api/zoom_eye'
+
+        summary "Queries the Domains https://api.zoomeye.hk API"
+        description <<~DESC
+          Queries the Domains https://api.zoomeye.hk API and returns subdomains
+          and ip addresses of the domain.
+        DESC
```

Comment:
Also mention the `ZOOM_EYE_API_KEY` environment variable here in another paragraph.